core/src/consensus.rs

```diff
@@ -432,6 +460,14 @@ impl Tower {
         bank_weights.pop().map(|b| b.2)
     }
 
+    pub fn adjust_lockouts_if_newer_root(&mut self, root_slot: Slot) {
+        let my_root_slot = self.lockouts.root_slot.unwrap_or(0);
+        if root_slot > my_root_slot {
```

Comment:
@t-nelson what happens in this case if there are votes in the tower that are still locked out past the snapshot root?

Aka you have a fork structure that looks like:
```
           0
         /    \
       2.    3
```

Most recently you voted for 2 in your tower, but your snapshot root is `3`, you shouldn't vote until at least slot 4. 

I thought that was why you had to consult the large set of ancestors in the snapshot root to see if you're still locked out/when you can start voting, or have we ditched that design?

Reply:
Yeah that sounds right.  IIRC we decided to punt those changes to a later PR, being as it's a corner case of a corner case.